src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -84,24 +85,83 @@ def api_endpoint(self) -> str:
         """
         raise NotImplementedError()
 
+    def exec_line(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+        enable_stream_output=False,
+    ) -> LineResult:
+        start_time = datetime.utcnow()
+        # call execution api to get line results
+        url = self.api_endpoint + "/execution"
+        payload = {"run_id": run_id, "line_number": index, "inputs": inputs}
+        headers = {"Accept": "text/event-stream"} if enable_stream_output else None
+
+        if enable_stream_output:
+
+            def generator():
+                with httpx.Client() as client:
+                    with client.stream(
+                        "POST", url, json=payload, timeout=LINE_TIMEOUT_SEC, headers=headers
+                    ) as response:
+                        if response.status_code != 200:
+                            run_info = FlowRunInfo.create_with_error(start_time, inputs, index, run_id, result)
+                            yield LineResult(output={}, aggregation_inputs={}, run_info=run_info, node_run_infos={})
+                        for line in response.iter_lines():
+                            chunk_data = json.loads(line)
+                            # only support one chat output for now
+                            yield LineResult.deserialize(chunk_data)
+
+            origin_generator = generator()
+            line_result = next(origin_generator)
+            chat_output_name = list(line_result.output.keys())[0]
```

Comment:
If response.status_code is not 200, the outputs of the line result will be empty and an IndexError will be raised here.

Reply:
handle here: 
![image](https://github.com/microsoft/promptflow/assets/26239730/f8f60a49-9514-4fed-b095-ba76a6b7ff8a)